code/ndarray.c

```diff
@@ -935,7 +935,33 @@ mp_obj_t ndarray_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw,
     mp_arg_check_num(n_args, n_kw, 1, 2, true);
     mp_map_t kw_args;
     mp_map_init_fixed_table(&kw_args, n_kw, args + n_args);
-    return ndarray_make_new_core(type, n_args, n_kw, args, &kw_args);
+    uint8_t dtype = ndarray_init_helper(n_args, args, &kw_args);
```

Comment:
Wouldn't it be simpler to move this whole block to `make_new_core`? That would solve the `circuitpytthon` problem.

Reply:
Sounds like a plan :snowboarder: